packages/rabbitmq/src/amqp/utils.ts

```diff
@@ -29,7 +29,7 @@ export function matchesRoutingKey(
 }
 
 const rabbitMQRegex =
-  /^amqps?:\/\/(([^:]+):([^@]+)@)?([^:/]+)(:[0-9]+)?(\/.*)?$/;
+  /^amqps?:\/\/(?:[^@]+@)?[^:/?#]+(?::\d+)?(?:\/[^?#]*)?(?:\?.*)?$/;
```

Comment:
actually, I thought in doing it in a simple way, just parsing it as a `URL` instance.

We could do something like:
```ts
try {
  const uri = new URL(uri);
  return uri.protocol.startsWith('amqp') || uri.protocol.startsWith('amqps')
} catch (e) {
  // ...
}
```

IMO is less complex than handling RegEx per se, what do you think?

Reply:
Sure, looks like a good alternative. I will review your approach once I have some time free today